portal/src/main/webapp/config_service.jsp

```diff
@@ -55,16 +31,161 @@ studies += "}";
+    
     oncoprintCustomDriverAnnotationBinaryMenuLabel: '<%=GlobalProperties.getBinaryCustomDriverAnnotationMenuLabel()%>',
     oncoprintCustomDriverAnnotationTiersMenuLabel: '<%=GlobalProperties.getTiersCustomDriverAnnotationMenuLabel()%>',
     oncoprintCustomDriverAnnotationDefault:'<%=GlobalProperties.enableDriverAnnotations()%>' !== "false", // true unless "false"
     oncoprintCustomDriverTiersAnnotationDefault:'<%=GlobalProperties.enableTiers()%>' !== "false", // true unless "false"
     oncoprintOncoKbHotspotsDefault:{"true":undefined, "false":"disable", "custom":"custom"}['<%=GlobalProperties.enableOncoKBandHotspots()%>'],
     oncoprintHideVUSDefault:'<%=GlobalProperties.hidePassengerMutations()%>' === "true", // false unless "true"
-    priorityStudies : <%=studies%>,
+    priorityStudies : <%=studies%>,  //done
     sessionServiceIsEnabled: '<%=GlobalProperties.getSessionServiceUrl()%>' !== "",
+    sessionServiceUrl: '<%=GlobalProperties.getSessionServiceUrl()%>',
     baseUrl:"<%=baseURL%>",
-    fontendCongfigOveride:JSON.parse('<%=GlobalProperties.getFrontendConfig()%>')
-});
+    fontendCongfigOveride:JSON.parse('<%=GlobalProperties.getFrontendConfig()%>'),
+    skinShowDataTabSets: (<%=GlobalProperties.showDataTab()%> === true),
+    skinShowNewsTab: (<%=GlobalProperties.showNewsTab()%> === true),
+    skinShowFAQSTab: (<%=GlobalProperties.showFaqsTab()%> === true),
+    skinShowToolsTab: (<%=GlobalProperties.showToolsTab()%> === true),
+    skinShowAboutTab: (<%=GlobalProperties.showAboutTab()%> === true),
+    skinShowWebAPITab: (<%=GlobalProperties.showWebApiTab()%> === true),
+    skinShowRmatLABTab : (<%=GlobalProperties.showRMatlabTab()%> === true),
+    skinShowTutorialsTab: (<%=GlobalProperties.showTutorialsTab()%> === true),
+    emailContactAddress: "<%=GlobalProperties.getContactEmailAddress()%>",
+    skinDocumentationBaseUrl: "<%=GlobalProperties.getBaseUrl()%>",
+    skinIsMarkdownDocumentation: (<%=GlobalProperties.isMarkdownDocumentation()%> === true),
+    skinFaqSourceURL: "<%=GlobalProperties.getFaqHtml()%>",
+    skinAboutSourceURL: "<%=GlobalProperties.getAboutHtml()%>",
+    skinNewsSourceURL: "<%=GlobalProperties.getNewsHtml()%>",
+    skinOQLSourceURL: "<%=GlobalProperties.getOqlHtml()%>",
+    googleAnalyticsProfile: "<%=GlobalProperties.getGoogleAnalyticsProfileId()%>",
+    
+   
+};
+--%>
+
+// an empty string means the property is listed but not assigned to in portal.props 
+// a null value means it was entirely missing 
+
+<%= request.getParameter("callback") %>({
+
+    <%
+    
+        // Set API root variable for cbioportal-frontend repo
+        String url = request.getRequestURL().toString();
+        String baseURL = url.substring(0, url.length() - request.getRequestURI().length()) + request.getContextPath();
+        baseURL = baseURL.replace("https://", "").replace("http://", "");
+    
+        String[] propNameArray = {
+            "app.version",
```

Comment:
we'll test these more thoroughly before release: https://github.com/cBioPortal/cbioportal/issues/4972